pydoctor/test/test_astbuilder.py

```diff
@@ -1988,3 +1988,16 @@ def test_constant_override_do_not_warns_when_defined_in_module_docstring(systemc
     assert ast.literal_eval(attr.value) == 99
     captured = capsys.readouterr().out
     assert not captured
+
+@systemcls_param
+def test__name__equals__main__is_skipped(systemcls: Type[model.System]) -> None:
+    mod = fromText('''
+    if __name__ == '__main__':
+        foo = True
```

Comment:
Maybe add a minimal external API to shows that the presence of `if __main__` only ignores the code inside the conditional block and does not ignores the whole module.

```suggestion
    PUBLIC_API = 'some value'
    if __name__ == '__main__':
        foo = True
```